docs/IMPLEMENTATION.md

```diff
@@ -0,0 +1,97 @@
+
+For the current design scenario of kata CC, only one KBC instance of AA on a k8s pod exists in the whole life cycle of the pod. This KBC instance will be created when AA receives the first UnWrapKey request. However, the implementation of KBC runtime provides good scalability for multiple KBC instances that may occur in other practice scenarios or in the future of kata CC.
+
+## Pass KBC name and KBS URI to AA
+
+In the current implementation, KBC name and KBS URI are passed by the implementer of key provider protocol, e.g, ocicrypt and ocicrypt-rs.
+
+### Cannot pass through layer annotation currently
+
+These two information should not be placed in the layer annotation as part of the container image. There are two main reasons: 
+
+1. **Flexibility**: not binding KBC name and KBS URI to container image can improve the cross platform portability of container image. (for example, suppose a KBC can only support the SEV platform. If the KBC name or KBS URI is written to the layer annotation when encrypting a container image, it means that the encrypted container image can only run on the SEV platform).
+2. **Security**: layer annotation is public plaintext data. Without additional encryption protection, an attacker can launch DoS attacks by using KBS URI to prevent the tenant from starting any confidential container based on the KBS; Without additional integrity protection, an attacker can tamper with the contents and induce potential security problems in the process of decrypting the image layer.
+
+**! ! ATTENTION ! ! : **However, it must be noted here that for the first reason mentioned above, we do not rule out the possibility that there may be KBCs supporting various platforms (EAA protocol KBC is trying to do so). In this case, placing KBC name in layer annotation will not affect the cross platform portability of container image, but will support the complex scenarios such as "different layers of a same container image use different KBS protocol to encrypt" more flexibly. Therefore, the current AA code implementation retains the scalability of multiple KBC instances at runtime in order to better support the above possible changes in the future.
+
+### Passed by ocicrypt instead of kata-agent
+
+In the implementation of AA, the KBC name and KBS URI are passed through the keyprovider protocol of ocicrypt, rather than by Kata agent in the startup phase of AA.
+
+KBC name::KBS URI is passed by the user-defined field reserved in the keyprovider protocol. The request of UnWrapKey API should be as follows: 
+
+```
+{
+    "op":"keyunwrap",
+    "keyunwrapparams":{
+        "dc":{
```

Comment:
Maybe we should add DC and EC to the glossary. I don't think the acronyms are explained anywhere in the code.

Reply:
OK, we have added a separate chapter in the IMPLEMENTATION.md document to describe the fields and their meanings in the keyprovider protocol.